packages/sessions-sdk-react/src/components/send-token-page.tsx

```diff
@@ -283,13 +294,24 @@ const SendTokenPageImpl = ({
         onSubmit: FormEventHandler;
         recipient: string;
         amount: string;
+        price: number | undefined;
         onChangeRecipient: (newRecipient: string) => void;
         onPressScanner: () => void;
         onChangeAmount: (newAmount: string) => void;
         feeConfig: Awaited<ReturnType<typeof getTransferFee>>;
       }
   )) => {
   const scannerShowing = !props.isLoading && props.scanner !== undefined;
+  const maxSendAmount = !props.isLoading && !props.isSubmitting && props.feeConfig.mint.equals(tokenMint)
+    ? amountAvailable - props.feeConfig.fee
+    : amountAvailable;
```

Comment:
This is fundamentally incorrect.  You have a risk that if a network connection is slow e.g. you will fall back to `amountAvailable` because of `isLoading`.  Currently the UI will prevent you from entering anything so I don't think you'll have bugs, but the code is very dangerous, because `maxSendAmount` is not valid or usable if `isLoading` or `isSubmitting` -- if some later refactor were to use `maxSendAount` in a place that _wasn't_ guarded by `isLoading`, this would lead to subtle bugs that are hard to understand and would not be caught if someone wasn't explicitly testing USDC.

I think if you want to do this, you need to think about restructuring the code a bit so you don't have `maxSendAmount` ever in a state where the value it holds is incorrect.